server/src/main/java/org/elasticsearch/rest/AbstractRestChannel.java

```diff
@@ -96,7 +98,11 @@ public XContentBuilder newBuilder(@Nullable XContentType requestContentType, boo
     public XContentBuilder newBuilder(@Nullable XContentType requestContentType, @Nullable XContentType responseContentType,
             boolean useFiltering) throws IOException {
         if (responseContentType == null) {
-            responseContentType = XContentType.fromMediaTypeOrFormat(format);
+            //TODO PG shoudld format vs acceptHeader be always the same, do we allow overriding?
```

Comment:
Maybe open an issue for this. My guess is that there is some precedence already and it might be that format overrides the header

Reply:
raised https://github.com/elastic/elasticsearch/issues/62294 to discuss this